openqaoa/optimizers/training_vqa.py

```diff
@@ -630,6 +637,15 @@ def optimize(self):
         elif self.method == 'spsa':
             print("Warning : SPSA is an experimental feature.")
             method = om.SPSA
+        elif self.method.lower().split('_')[0] == 'pennylane': # check if we are using a pennylane optimizer
```

Comment:
I see that Pennylane optimisers are integrated under the `CustomScipyGradientOptimizer` class. Is there a reason we don't use a new class entirely instead? (say, `PennelaneOptimizer`, with a corresponding list of methods `PENNYLANE_OPTIMIZERS = ['pennylane_adagrad', ... ]`)